File: src/ReverseSentence.cpp
class Solution {
public:
    string ReverseSentence(string str) {
      string &tmpStr = str;

    }


    void Reverse(string& str, int start, int end) {
      if (start > end || str.size() == 0) {
        return;
      }
      while (start < end) {
        char temp = str[start];
        str[start] = str[end];
        str[end] = temp;

        start++;
        end--;
      }
    }
};

File: src/java/ReverseSentence.java
public class Solution {
    public String ReverseSentence(String str) {
        // only blanks
        if (str.trim().equals("")) {
            return str;
        }

        // reverse the whole string
        String revString = reverse(str);

        // split the sentence
        String[] splitStrings = revString.split(" ");

        String result = "";

        // reverse each word
        for (String s : splitStrings) {
            result += reverse(s) + " ";
        }

        return result.trim();
    }

    // reverse a string
    private String reverse(String str) {
        char[] a = str.toCharArray();
        char temp;
        int start = 0;
        int end = a.length - 1;
        while (start < end) {
            temp = a[start];
            a[start] = a[end];
            a[end] = temp;

            start++;
            end--;
        }
        return String.valueOf(a);
    }
}
